test/ProtocolRewardsTest.sol

```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+import "forge-std/Test.sol";
+
+import "../contracts/ProtocolRewards.sol";
+
+contract ProtocolRewardsTest is Test {
+    uint256 internal constant ETH_SUPPLY = 120_200_000 ether;
```

Comment:
I have a preference for hard-coding all magic numbers as constants like this, but it's not mission critical. Let's you use more relative statements and makes conversion to fuzz testing a little smoother. Mainly relevant for the little `5 ether`, `10 ether` bits and bobs in various tests.